src/utils/formUtils.js

```diff
@@ -1,47 +1,97 @@
+        pageLayout: form.get("pagelayout"),
+        cropMarks: form.has("cropmarks"),
+        cutMarks: form.has("cutmarks"),
+        pageScaling: form.get("page_scaling"),
+        pagePositioning: form.get("page_positioning"),
+        mainForeEdgePaddingPt: form.get("main_fore_edge_padding_pt"),
+        bindingEdgePaddingPt: form.get("binding_edge_padding_pt"),
+        topEdgePaddingPt: form.get("top_edge_padding_pt"),
+        bottomEdgePaddingPt: form.get("bottom_edge_padding_pt"),
+        sigFormat: form.get("sig_format"),
+        sigLength: form.get("sig_length"),
+        customSigLength: form.get("custom_sig"),
+        foreEdgePaddingPt: form.get("fore_edge_padding_pt"),
+        wackySpacing: form.get("wacky_spacing"),
+        fileDownload: form.get("file_download"),
+        printFile: form.get("print_file"),
+        flyleaf: form.has("flyleaf"),
+        paperSizeCustomWidth: form.get("paper_size_custom_width"),
+        paperSizeCustomHeight: form.get("paper_size_custom_height"),
+    });
+
+/**
+ * Sets the configuration to the URL.
+ * @param { import("../models/configuration").Configuration } configuration The configuration to set
+ */
+const setConfigurationToUrl = (configuration) => {
+    updateWindowLocation(setUrlParams(window.location.href, configuration));
+};
+
+/**
+ * Loads settings from the URL or local storage.
+ * @returns { import("../models/configuration").Configuration } The configuration
+ */
+const loadConfiguration = () => {
+    const urlParams = toUrlParams(window.location.href);
+    const hasUrlParams = Object.keys(urlParams).length > 0;
+
+    const localSettings = hasUrlParams ? urlParams : getLocalSettings();
+
+    const configuration = schema.parse(localSettings);
+    setConfigurationToUrl(configuration);
+    return configuration;
+};
+
+/**
+ * Updates the rendered form from a book.
+ * @param { import("../book").Book } book The book to update the form from
+ */
+export function updateRenderedForm(book) {
+    console.log("Form updated....");
     book.createpages().then(() => {
-        console.log("... pages created")
+        console.log("... pages created");
         renderPageCount(book);
         renderInfoBox(book);
         renderWacky();
-    })
+    });
 }
 
-export function saveForm(formData, book) {
-    const bookSettings = {
-        duplex: book.duplex,
-        duplexrotate: book.duplexRotate,
-        format: book.format,
-        sigsize: book.sigsize,
-        lockratio: book.lockratio,
-        papersize: formData.get('paper_size'),
-        pagelayout: formData.get('pagelayout'),
-    };
-    setLocalSettings(bookSettings);
+/**
+ * Saves a form and updates the service's configuration.
+ * @param { FormData } form The form to save
+ * @returns { import("../models/configuration").Configuration } The updated configuration set
+ */
+export function saveForm(form) {
+    const configuration = fromFormToConfiguration(form);
+    setLocalSettings(configuration);
```

Comment:
I'm not wholly opposed to setting everything into local store, but we probably need to have a quick group discussion to make sure there aren't any options that are going to drive people nuts when they're set every time they visit the site, particularly when there's no way to quickly clear all the settings (maybe a good first ticket for someone else to add?)